### ISSUE TITLE
Constructor annotated with any type other than the default qualifier not allowed

### ISSUE THREAD

**Issue by t-rasmud** [typetools/checker-framework#2186]
Annotating a class with a non-default type qualifier results in an unexpected error on its constructor.
Example:
```
import org.checkerframework.checker.propkey.qual.*;
@PropertyKey class ConstructorIssue {
    ConstructorIssue() {}
}
```
The above code should type check without errors and the type of the constructor should be `@PropertyKey`. But, running the command: `javac -processor propertykey ConstructorIssue.java ` gives the following error:

```
ConstructorIssue.java:4: error: [type.invalid.annotations.on.use] invalid type: annotations [@UnknownPropertyKey] conflict with declaration of type ConstructorIssue
    ConstructorIssue() {}
    ^
1 error
```
